akka-projection-core/src/main/scala/akka/projection/internal/InternalProjectionState.scala

```diff
@@ -375,12 +441,22 @@ private[akka] abstract class InternalProjectionState[Offset, Envelope](
           .andThen { case _ => handlerLifecycle.tryStop() }
           .andThen {
             case Success(_) =>
+              telemetry.stopped()
               statusObserver.stopped(projectionId)
             case Failure(AbortProjectionException) =>
+              telemetry.stopped()
               statusObserver.stopped(projectionId) // no restart
             case Failure(exc) =>
-              Try(statusObserver.stopped(projectionId))
-              statusObserver.failed(projectionId, exc)
+              // For observer and telemetry, invoke failed first and stopped second.
+              // Propagate the exception thrown by statusObserver.failed only (swallowing
+              // any other exception).
+              try {
+                Try(telemetry.failed(exc))
+                Try(telemetry.stopped())
+                statusObserver.failed(projectionId, exc)
+              } finally {
+                Try(statusObserver.stopped(projectionId))
```

Comment:
I don't know why the original `statusObserver.failed` wasn't wrapped with `Try`. Should be more interesting to propagate the `exc`. Also easier to read:
 ```
Try(telemetry.failed(exc))
Try(telemetry.stopped())
Try(statusObserver.failed(projectionId, exc))
Try(statusObserver.stopped(projectionId))
```